src/box/box.cc

```diff
@@ -4312,12 +4312,27 @@ box_process_fetch_snapshot(struct iostream *io,
 			  "wal_mode = 'none'");
 	}
 
-	say_info("sending current read-view to replica at %s", sio_socketname(io->fd));
-
+	/** Used for checkpoint initial join. */
+	struct engine_checkpoint_cursor cursor = {
+		.is_checkpoint_join = req.is_checkpoint_join,
+		.vclock = &req.checkpoint_vclock,
+		.last_lsn = req.checkpoint_lsn,
+		.lsn_counter = 0,
+	};
+	struct gc_checkpoint_ref gc;
+	auto gc_guard = make_scoped_guard([&] {
+		gc_unref_checkpoint(&gc);
+	});
+	gc_guard.is_active = false;
+	if (cursor.is_checkpoint_join) {
+		if (relay_prepare_checkpoint_for_join(&cursor, &gc) != 0)
+			diag_raise();
```

Comment:
Referencing the last checkpoint is really trivial. Let's implement it  completely in CE, without using the stubs, and let the engines raise the `ER_UNSUPPORTED` error in CE.

Reply:
I managed to properly split CE and EE part without moving "relay" part to CE